### PR TITLE
Achievements and score rebalance

### LINKED ISSUE #8: Add streak counter
At minimum there should be 3 interactions that keep track if there is a streak for the interaction. These should tie directly to atleast 3 achievments associated with the streaks

### LINKED ISSUE #9: Add Achievments
There should be at minimum 3 achievments that tie directly to 3 seperate obtain streaks through interactions. Notify the player when an achievment is obtained and reward it as bonus points in scoreboard calculation.

### PR DESCRIPTION
**Pull request by cooperjl**
Closes #8, closes #9.

Rebalances a few different activities around the map and changes the scoring algorithm.

### CODE REVIEW

File: core/src/uk/ac/york/student/game/activities/Streak.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+/**
+ * Class which represents a streak of activities.
+ */
+@Getter
+public class Streak {
+    /**
+     * A Boolean value used to lock the value of streak for if the player does the activity twice in one day.
+     */
+    Boolean daily;
```
Comment (tandemdude) [inline core/src/uk/ac/york/student/game/activities/Streak.java]: Unless this is intended to be nullable you should probably specify the type as `boolean` not `Boolean`

File: core/src/uk/ac/york/student/screens/GameScreen.java [hunk 2]
```diff
@@ -866,16 +855,34 @@ private boolean doActivity(@NotNull ActivityMapObject actionMapObject) {
...
+            float changeAmount = actionMapObject.getChangeAmount(metricType);
+            // Apply the effect to the metric
+            metrics.changeMetric(metricType, metricEffect, changeAmount);
+        }
+        // Increment the streak
+        if (type == Activity.STUDY) {
+            player.getStreaks().increaseStreak("study");
+        } else {
+            player.getStreaks().increaseStreak(actionMapObject.getStr());
+        }
```
Comment (tandemdude) [inline core/src/uk/ac/york/student/screens/GameScreen.java]:
```suggestion
        player.getStreaks().increaseStreak(type == Activity.STUDY ? "study" : actionMapObject.getStr());
```

File: core/src/uk/ac/york/student/score/ScoreManager.java [hunk 3]
```diff
@@ -76,19 +77,21 @@ public static List<Integer> getTop10Scores() {
...
+        float metricWeighting = (maxScore - percentForStreaks) / (maxHappiness + maxStudyLevel);
 
-        float totalScore = energyScore + studyScore + happinessScore;
-        float maxPossibleScore = energyWeighting + studyWeighting + happinessWeighting;
+        float achievementsScore = achievements.size() * achievementsWeighting;
+        float studyScore = studyLevel * metricWeighting;
+        float happinessScore = happiness * metricWeighting;
 
-        return (totalScore / maxPossibleScore) * 100;
+        return studyScore + happinessScore + achievementsScore;
```
Comment (tandemdude) [inline core/src/uk/ac/york/student/score/ScoreManager.java]: Have you tested that this scales between 0 and 100 - that's what the score to string method expects.

File: core/src/uk/ac/york/student/game/activities/Streak.java (hunk 1 above)
Comment (cooperjl) [reply core/src/uk/ac/york/student/game/activities/Streak.java]: Fixed

File: core/src/uk/ac/york/student/score/ScoreManager.java (hunk 3 above)
Comment (cooperjl) [reply core/src/uk/ac/york/student/score/ScoreManager.java]: The maths should be right and works fine, but added way to ensure it is just in case.

**Pull request merged**